test-framework/sudo-compliance-tests/src/lecture.rs

```diff
@@ -0,0 +1,101 @@
+use sudo_test::{Command, Env, User};
+use crate::{Result, SUDOERS_ROOT_ALL, USERNAME, SUDOERS_USER_ALL_ALL, SUDOERS_ALWAYS_LECTURE, SUDOERS_NO_LECTURE, PASSWORD, OG_SUDO_STANDARD_LECTURE};
+
+#[test]
+#[ignore]
+fn default_lecture_shown_once() -> Result<()> {
+    let env = Env([SUDOERS_ROOT_ALL, SUDOERS_USER_ALL_ALL])
+        .user(User(USERNAME).password(PASSWORD))
+        .build()?;
+
+    let output = Command::new("sudo")
+    .args(["-S", "true"])
+    .as_user(USERNAME)
+    .stdin(PASSWORD)
+    .exec(&env)?;
+    assert_eq!(true, output.status().success());
+
+    assert_contains!(
+        output.stderr(),
+        OG_SUDO_STANDARD_LECTURE
+    );
+
+    let second_sudo = Command::new("sudo")
+    .as_user(USERNAME)
+    .stdin(PASSWORD)
+    .args(["-S", "echo", "Yeah!"])
+    .exec(&env)?;
+
+    assert_eq!(true, second_sudo.status().success());
+    assert_ne!(second_sudo.stdout().unwrap(), OG_SUDO_STANDARD_LECTURE);
```

Comment:
this should check stderr instead of stdout. this should use `assert_not_contains` because the stderr also contains the password prompt so `assert_ne!` will accept the case where the lecture is shown (before the prompt)

Reply:
I think it would be worthwhile to have another test that checks that the lecture is printed to stderr and *not* to stdout. so, that should contain a `assert_contains!(stderr, LECTURE)` *and* `assert_not_contains!(stdout, LECTURE)`